microcosm_sqlite/stores.py

```diff
@@ -158,8 +158,14 @@ def search(self, **kwargs):
 
         """
         query = self._query()
-        query = self._order_by(query, **kwargs)
         query = self._filter(query, **kwargs)
+        query = self._order_by(query, **kwargs)
+
+        if offset is not None:
+            query = query.offset(offset)
+        if limit is not None:
+            query = query.limit(limit)
```

Comment:
Seems like we could easily get this in `count`, `first` et al. by putting those four lines in a `_paginate` method.
Also it may be worth putting your explanation of the need for this particular ordering in a comment somewhere.